### PR TITLE
fix!: split out upgrade into a separate signal module

### PR DESCRIPTION
**Pull request by cmwaters**
Closes: https://github.com/celestiaorg/celestia-app/issues/3284

We've been experienceing an app hash mismatch when running `main` on mainnet. I believe this is caused because in v1 we had an upgrade module (albeit unreachable) which was still part of state (kept around for compatibility with IBC). In `main` we coped the upgrade modules state and extended it with the signalling mechanism however this module only applied from v2 to v2 meaning there wasn't the same state in v1. 

This attempts to fix this. It separates out the signal protocol into it's own module (which correctly should only be v2) and reimport the upgrade keeper from the sdk (as was in v1.x)

### CODE REVIEW

**Comment by cmwaters** [comment]
I will try run this on `mainnet` to see if the app hash persists

**Comment by cmwaters** [comment]
In v1, we registered all the upgrade types in the encoding register:

https://github.com/celestiaorg/celestia-app/blob/91b6827a80ef8a4d0048ca81f9eda4c88f37a2ef/app/app.go#L163

These are types like `upgrade.Plan` which don't and shouldn't be used. I am wondering if we should still register them

**Comment by cmwaters** [comment]
This hasn't completely fixed the app hash mismatch problem but I think this change is still important.

What nina and I have noticed is that there is a mismatch in the store keys between v1.x and main (which is the source for how the app hash is created). So what it seems we will need to do is:
1) add back the v1.x keys (i.e. blobmodule keys)
2) remove the new keys and instead add the keys to the store during the upgrade by calling `LoadVersionAndUpgrade` in the `MultiCommitStore`

File: x/signal/README.md [hunk 1]
```diff
@@ -27,18 +27,18 @@ See [types/msgs.go](./types/msgs.go) for the message types.
 
 ```shell
 celestia-appd query upgrade tally
```
Comment (rootulp) [inline x/signal/README.md:29]: I assume the query command is also moving from the upgrade module to the signal module

```suggestion
celestia-appd query signal tally
```

File: x/signal/keeper.go [hunk 2]
```diff
@@ -20,12 +20,12 @@ var (
 	defaultSignalThreshold = sdk.NewDec(5).Quo(sdk.NewDec(6))
 )
 
-// SignalThreshold is the fraction of voting power that is required
+// Threshold is the fraction of voting power that is required
```
Comment (rootulp) [inline x/signal/keeper.go:23]: 👍

File: x/signal/keeper_test.go [hunk 3]
```diff
@@ -20,6 +24,52 @@ import (
 	tmdb "github.com/tendermint/tm-db"
 )
 
+func TestGetVotingPowerThreshold(t *testing.T) {
```
Comment (rootulp) [inline x/signal/keeper_test.go:27]: Thanks for moving tests to `keeper_test.go`! I just find it easier to find the test if a function is defined in foo.go then the unit test is in foo_test.go.

I'll revert the change from https://github.com/celestiaorg/celestia-app/pull/3277 and/or rebase after this merges

File: x/signal/types/msgs.go [hunk 4]
```diff
@@ -5,23 +5,17 @@ import (
...
 	QuerierRoute = ModuleName
+	RouterKey    = ModuleName
 
-	// RouterKey defines the module's message routing key
-	RouterKey = ModuleName
-
-	URLMsgSignalVersion = "/celestia.upgrade.v1.Msg/SignalVersion"
-	URLMsgTryUpgrade    = "/celestia.upgrade.v1.Msg/TryUpgrade"
+	URLMsgSignalVersion = "/celestia.signal.v1.Msg/SignalVersion"
+	URLMsgTryUpgrade    = "/celestia.signal_test.v1.Msg/TryUpgrade"
```
Comment (rootulp) [inline x/signal/types/msgs.go:18]: [blocking]

```suggestion
	URLMsgTryUpgrade    = "/celestia.signal.v1.Msg/TryUpgrade"
```

**Review by rootulp — COMMENTED** [review]
I think it's okay if proto-check-breaking fails [here](https://github.com/celestiaorg/celestia-app/actions/runs/8628726730/job/23651430463?pr=3290#step:3:19) because we haven't cut a release from main yet.

One blocking comment but the rest LGTM. Thanks for diving deep on this investigation!

File: x/signal/types/msgs.go (hunk 4 above)
Comment (cmwaters) [reply x/signal/types/msgs.go:18]: good catch. Don't know how I changed that.

**Review by rootulp — APPROVED** [review]
I just remembered we may need to modify the specs to include a link to the new signal module but that doesn't strictly need to happen in this PR

**Comment by cmwaters** [comment]
> I just remembered we may need to modify the specs to include a link to the new signal module but that doesn't strictly need to happen in this PR

good point

**Review by evan-forbes — APPROVED** [review]
I'm _signaling_ my approval of this PR

**Pull request merged**